services/src/main/scala/cromwell/services/metadata/impl/MetadataServiceActor.scala

```diff
@@ -28,7 +29,14 @@ object MetadataServiceActor {
 
 case class MetadataServiceActor(serviceConfig: Config, globalConfig: Config)
   extends Actor with ActorLogging with MetadataDatabaseAccess with SingletonServicesStore {
-
+  
+  override val supervisorStrategy = new OneForOneStrategy()(SupervisorStrategy.defaultStrategy.decider) {
+    override def logFailure(context: ActorContext, child: ActorRef, cause: Throwable, decision: Directive) = {
+      val childName = if (child == readActor) "Read" else "Write"
+      log.error(s"The $childName Metadata Actor died unexpectedly. Restarting it...", cause)
```

Comment:
What I mean is this. Whatever reason caused the WMA to crash, it certainly wasn't the accumulated queue of events which have built up after it decided to start writing to DB, so there's no reason we should be losing those. Finding a way to preserve that queue would be helpful (although it still wouldn't protect in the case that the DB was down and everything fails, but that's a separate topic).